modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -954,7 +954,40 @@ def median(self, axis, **kwargs):
             return self.default_to_pandas(pandas.DataFrame.median, axis=axis, **kwargs)
         return Reduce.register(pandas.DataFrame.median)(self, axis=axis, **kwargs)
 
-    nunique = Reduce.register(pandas.DataFrame.nunique)
+    def nunique(self, axis=0, dropna=True):
+        if not RangePartitioning.get():
+            return Reduce.register(pandas.DataFrame.nunique)(
+                self, axis=axis, dropna=dropna
+            )
+
+        unsupported_message = ""
+        if axis != 0:
+            unsupported_message += (
+                "Range-partitioning 'nunique()' is only supported for 'axis=0'.\n"
```

Comment:
Is it difficult to support axis=1?

Reply:
In the current implementation of range-partitioning mechanism, it's impossible. Rewriting the whole mechanism just for this method doesn't seem reasonable